src/compiler/checker.ts

```diff
@@ -8540,6 +8550,18 @@ namespace ts {
             return undefined;
         }
 
+        function getImmediatelyInvokedFunctionExpression(func: FunctionExpression | MethodDeclaration) {
+            if (isFunctionExpressionOrArrowFunction(func)) {
+                let parent = func.parent;
+                while (parent.kind === SyntaxKind.ParenthesizedExpression) {
+                    parent = parent.parent;
+                }
+                if (parent.kind === SyntaxKind.CallExpression) {

```

Comment:
i think this is what is causing the other issue you mentioned. we are not checking for func being in the target of the call and not an argument.